mypostcard/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bytes"
+	"embed"
+	"flag"
+	"fmt"
+	"image"
+	"image/draw"
+	"image/png"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/freetype/truetype"
+	"golang.org/x/image/font"
+	"golang.org/x/image/math/fixed"
+)
+
+//go:embed LICENSE
+var license string
+
+//go:embed postcard.png
+var postcardPNG []byte
+
+//go:embed stamp*.png
+var stampsPNG embed.FS
+
+//go:embed ClarendonBT.ttf
+var clarendonBT []byte
+
+//go:embed *.html *.css LICENSE
+var webUI embed.FS
+
+func main() {
+	listenAddr := flag.String("l", ":8000", "Listen address")
+	printLicense := flag.Bool("license", false, "Print license")
+	flag.Parse()
+	if *printLicense {
+		fmt.Println(license)
+		return
+	}
+	rand.Seed(time.Now().UnixNano())
+	http.Handle("/", http.FileServer(http.FS(webUI)))
+	http.HandleFunc("/postcard", postcardHandler)
+	fmt.Println("Listen and serve on", *listenAddr)
+	err := http.ListenAndServe(*listenAddr, nil)
+	if err != nil {
+		fmt.Println("ERROR", err)
+	}
+}
+
+func postcardHandler(w http.ResponseWriter, r *http.Request) {
+	text := r.FormValue("postcard")
+	text = strings.ReplaceAll(text, "\r", "")
```

Comment:
nit: extract to a func with proper name.

Reply:
I'll keep it as one func - a lot of the code here could be reused or extracted into functions but wanted each part as a function for the blog post.